### PR TITLE
Use per dr gasprice

### PR DESCRIPTION
**Pull request by girazoki**

### CODE REVIEW

File: bridges/ethereum/src/actors/claim_and_post.rs [hunk 1]
```diff
@@ -314,7 +314,32 @@ fn try_to_claim_local_query(
...
+                        .query(
+                            "readGasPrice",
+                            dr_id,
+                            eth_account,
+                            contract::Options::default(),
+                            None,
+                        )
+                        .map_err(move |e| {
+                            log::warn!(
+                                "[{}] Error in params reception:  {}",
```
Comment (tmpolaczyk) [inline bridges/ethereum/src/actors/claim_and_post.rs:332]: Mention `readGasPrice` in this error

File: bridges/ethereum/src/actors/block_relay_and_poi.rs [hunk 2]
```diff
@@ -212,13 +215,22 @@ pub fn block_relay_and_poi(
...
                     let eth_state = Arc::clone(&eth_state);
                     move |(including, resolving, needs_relaying)| {
-
                         if (is_new_block && config.enable_block_relay_new_blocks) || (!is_new_block && config.enable_block_relay_old_blocks) {
                             // Optimization: do not process blocks that do not contain requests coming from ethereum
                             if including.is_empty() && resolving.is_empty() && !needs_relaying {
                                 log::debug!("Skipping empty superblock");
                                 return futures::finished(());
                             }
+                            let last_id =  if let Some(&(id, _, _, _, _)) = including.last() {
```
Comment (tmpolaczyk) [inline bridges/ethereum/src/actors/block_relay_and_poi.rs:224]: There is a special syntax to "ignore the other fields"

```suggestion
                            let last_id =  if let Some(&(id, ..)) = including.last() {
```

File: bridges/ethereum/src/actors/block_relay_and_poi.rs [hunk 3]
```diff
@@ -212,13 +215,22 @@ pub fn block_relay_and_poi(
...
+                                Some(id)
+                            }
+                            else if let Some(&(id, _, _, _, _, _)) = resolving.last() {
```
Comment (tmpolaczyk) [inline bridges/ethereum/src/actors/block_relay_and_poi.rs:227]:
```suggestion
                            else if let Some(&(id, ..)) = resolving.last() {
```

File: bridges/ethereum/src/actors/block_relay_and_poi.rs [hunk 4]
```diff
@@ -236,29 +248,49 @@ pub fn block_relay_and_poi(
...
-                                            block_relay_contract2
-                                                .call_with_confirmations(
-                                                    "postNewBlock",
-                                                    (superblock_hash, superblock_epoch, dr_merkle_root, tally_merkle_root),
-                                                    eth_account,
-                                                    contract::Options::with(|opt| {
-                                                        opt.gas = config.gas_limits.post_new_block.map(Into::into);
-                                                    }),
-                                                    1,
+                                            if let Some(last_id) = last_id {
```
Comment (tmpolaczyk) [inline bridges/ethereum/src/actors/block_relay_and_poi.rs:254]: I would add a comment here, something like:

    // Use the same gas price as the last request from this block.
    // If this block does not contain any requests coming from ethereum, default to "None" which will estimate the gas price using the ethereum client.

File: bridges/ethereum/src/actors/claim_and_post.rs [hunk 5]
```diff
@@ -303,18 +300,42 @@ fn try_to_claim_local_query(
...
+                )
+            }
+        })
+        .and_then({
+            let eth_state = Arc::clone(&eth_state);
+
+            move |(poe, sign_addr, witnet_pk, dr_output, u_point, v_point)| {
+                Box::new(
+                    eth_state.wrb_contract
+                        .query(
```
Comment (tmpolaczyk) [inline bridges/ethereum/src/actors/claim_and_post.rs:320]: use `get_gas_price` function?

File: bridges/ethereum/src/actors/block_relay_and_poi.rs [hunk 6]
```diff
@@ -1,7 +1,12 @@
 //! Actor which receives Witnet superblocks, posts them to the block relay,
 //! and sends proofs of inclusion to Ethereum
 
-use crate::{actors::handle_receipt, actors::WitnetSuperBlock, config::Config, eth::EthState};
+use crate::{
+    actors::handle_receipt,
+    actors::WitnetSuperBlock,
```
Comment (lrubiorod) [inline bridges/ethereum/src/actors/block_relay_and_poi.rs:6]: fmt :P

Comment (girazoki) [reply bridges/ethereum/src/actors/block_relay_and_poi.rs:6]: thx

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**